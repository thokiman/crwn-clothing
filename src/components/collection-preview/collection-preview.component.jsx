import React from "react";
import "./collection-preview.styles.scss";
import CollectionItem from "../collection-item/collection-item.component";

// Decide on Component 2, Homepage, Reusable Component
//pipeline 1->2.2

const CollectionPreview = ({ id, title, routeName, items }) => {
  return (
    <div className="collection-preview">
      <h1 className="title">{title.toUpperCase()}</h1>
      <div className="preview">
        {items
          .filter((item, idx) => idx < 4)
          .map(({ id, ...otherItemProps }) => {
            return <CollectionItem key={id} {...otherItemProps} />;
          })}
      </div>
    </div>
  );
};

export default CollectionPreview;
