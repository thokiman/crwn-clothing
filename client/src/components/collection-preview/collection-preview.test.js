import React from "react";
import { shallow } from "enzyme";
import { CollectionPreview } from "./collection-preview.component";

describe("CollectionPreview component", () => {
  let wrapper;
  let mockHistory;
  let mockMatch;
  const mockRouteName = "hats";
  beforeEach(() => {
    mockHistory = {
      push: jest.fn(),
    };
    mockMatch = {
      path: jest.fn(),
    };
    const mockProps = {
      title: "hats",
      items: [],
      history: mockHistory,
      match: mockMatch,
      routeName: mockRouteName,
    };
    wrapper = shallow(<CollectionPreview {...mockProps} />);
  });

  it("should render CollectionPreview", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should call history.push with the right string when TitleContainer clicked", () => {
    wrapper.find("TitleContainer").simulate("click");
    expect(mockHistory.push).toHaveBeenCalledWith(
      `${mockMatch.path}/${mockRouteName}`
    );
  });
});
