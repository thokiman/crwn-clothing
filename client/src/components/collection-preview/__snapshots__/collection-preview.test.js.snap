// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview component should render CollectionPreview 1`] = `ShallowWrapper {}`;
